lib/store.js

```diff
@@ -1,24 +1,38 @@
+import { lru } from 'tiny-lru'
+import { DEFAULT_OPTIONS } from './constants.js'
 export class Store {
-  constructor(timeWindowMs) {
+  constructor(
+    timeWindowMs,
+    intervalTimeExpiredKeysEvictionMs,
+    cacheSize = DEFAULT_OPTIONS.cacheSize
+  ) {
     this.timeWindowMs = timeWindowMs
-    this.resetStore()
-    this.interval = setInterval(() => this.resetStore(), timeWindowMs)
-  }
-
-  resetStore() {
-    this.hits = {}
+    this.cache = lru(cacheSize, timeWindowMs)
+    this.interval = setInterval(() => {
+      this.cache.keys().forEach(key => {
+        this.cache.get(key) // it deletes expired keys under the hood and returns undefined
```

Comment:
Why don't use a `delete`,  `remove` or `clear` (or similar) method to be more semantic? It's pretty weird using a `get` function as deleting, if one day this behavior has been changed will crash the code (making the code very sensible).

Reply:
There is no other way to delete the key when it is expired. Otherwise we will have to use `items` property directly which is not exposed.